src/lib/Iex/IexBaseExc.h

```diff
@@ -57,60 +57,60 @@ IEX_INTERNAL_NAMESPACE_HEADER_ENTER
 // Our most basic exception class
 //-------------------------------
 
-class BaseExc: public std::exception
+class IEX_EXPORT BaseExc: public std::exception
```

Comment:
A few years ago we painstakingly went from this form to the individual export form, because classes contained things, such as std::string that are not adorned for export. This led to subtle corruption style bugs when using OpenEXR as a DLL. One of the root causes of the nightmare was that BaseExc was sublcassed from std::string. I'm extremely nervous about undoing this change! Are any warnings about "some members are not decorated for export" emitted during compilation? One of the many symptoms was that exception handling never worked on Windows; OpenEXR exceptions segfaulted.

Reply:
Another issue that resulted was that symbol tables become ungodly huge, base exporting the entire class exports way more than you would expect, including symbols for any std objects within the class. Perhaps Microsoft has refined things and it's no longer the can of worms that it was? ... nervous,,,, it took a long time to figure out why Windows was so unstable and correct it....

Reply:
Well, you should never believe me about Windows things, but I'm quite certain that both OIIO and OSL use this method of tagging the whole class as exportable, and not the individual methods therein, and I have not had any reported problems. Maybe this was an issue that was fixed in MSVS long ago?

Reply:
no warnings I've seen on any of the platform compiles about things not adorned for export on any of the CI builds, but I think the failure on the mac is due to me missing something about that for that platform - exceptions are notoriously difficult in this realm, so I think one / some of the Imf exceptions are not being exported correctly.

Reply:
We recently did a pass on USD to do it the pain-staking member-by-member way, rather than class by class, because we ran into, in particular, that since we have vector<string>, vector<string> was being re-exported as somehow being owned by USD libs. The root problem boils down to STL classes being re-exported by client libraries, and the fact that historically the Microsoft STL relied on static member initialization which breaks down fiendishly in a DLL based world. A quick scan of https://github.com/microsoft/STL suggests that they no longer rely on static initializers. Still, there may be a symbol table explosion due to undesired STL exports that the painstaking method avoids. At ILM, the historic motivation to do it the nasty way was that between boost and the STL the Zeno symbol table became too huge and we had to do fairly frequent refactorings to get templated symbols below the MSVC crash threshold.

All this is to say that I'm inclined to the simpler suggestion  you're proposing, if the symbol table sizes don't increase by integer multiples of the current size, and if exceptions thrown under Windows don't go back to segfaulting.